REV - Review of metadata general requirements. (#484)
DILCISBoard/E-ARK-CSIP: specification/implementation/metadata/general-requirements/index.md

## 5.1. General requirements for metadata in a CSIP Information Package

> The primary consideration when considering metadata requirements is, as with the rest of this specification, that of interoperability. That is to say that we focus on high-level technical interoperability and tasks which allow an Information Package to be prepared, transferred and received regardless of the institutions and tools involved. These tasks include:

Reviewer: ```suggestion
The primary consideration with metadata is, as with the rest of this specification, interoperability. Specifically, the high-level technical interoperability and tasks that allow an Information Package to be prepared, transferred and received, regardless of the institutions or tools involved. These tasks include:
```

- Identifying uniquely an Information Package and its components;
- Validating an Information Package;

> - Validating the contents of an Information Package;

Reviewer: >- Validating an Information Package;
>- Validating the contents of an Information Package;

These two are very similar - what is the difference?

> - Establishing the authenticity of the Information Package;

Reviewer 2: I am not a native English speaker, so disregard if my understanding is not correct, but I understand the word "establish" to be more "active", and "moderating", and "creating something new" and even though we as archives are and must be active in order to do digital preservation, then I read this sentence as if we create the authenticity. I understand our role and tasks more as we establish the organization around digital objects in order to make sure their authenticity will remain intact.

Reviewer: I agree that the archives should absolutely not "create authenticity" but preserve the authenticity that we are given. And for that, we need to find out exactly how much authenticity there is in the IP, so we need to test, analyse, validate, determine, examine. "Establish" is a better word for that than "prove," as prove implies convincing someone else, while here the goal is to find out the truth for ourselves.

The bullets here are a list of "tasks that allow an IP to be prepared, transferred and received," which is very technical, so I think "validate the authenticity of the IP" is the most accurate, although it repeats "validate."

Author: Ok, so I was going for an "establishing the facts" style use, am tempted to stick with the wording. Would like to avoid overloading validation.

- Accessing the contents of an Information Package.

In technical terms, the CSIP tries to control metadata needed by tools or users to:

- navigate data and metadata components within the package (i.e. packaging metadata);
- to validate that no component has been damaged during transfer or preservation (i.e. fixity information);
- to understand the processes used when creating and managing the package (i.e. provenance and preservation metadata); and
- to understand how the data within the package can be accessed (i.e. representation information).

> We regard descriptive metadata and most detailed technical metadata as outside the scope of the CSIP. As such, the CSIP does not aim to provide detailed semantic interoperability between different systems. However, as noted in Section 1.2, implementers are welcome to use the Content Information Type Specifications to achieve a  higher level of interoperability.

Reviewer: ```suggestion
We regard descriptive metadata and most of the detailed technical metadata as outside the scope of the CSIP. As such, the CSIP does not aim to provide detailed semantic interoperability between different systems. However, as noted in Section 1.2, implementers are welcome to use the Content Information Type Specifications to achieve interoperability on a more detailed level.
```

Reviewer: We use the vertical metaphor so that "high" means "general" and "low" means "detailed," e.g "high-level technical interoperability" in the 1st paragraph of § 5.1, Figure 3 in § 1.2, Figure 4 in § 1.3. So here it should be "lower level" or "deeper" interoperability, or simply "on a more detailed level."

> We describe the core metadata requirements using the METS (Metadata Encoding & Transmission Standard, <http://www.loc.gov/standards/mets/>). We describe the core elements used, the METS standard describes other available elements which may be used in addition to those detailed in the CSIP.

Reviewer 2: "the" METS - or only METS?

> Some metadata requirements are implied by the structure presented in the previous Section. As seen, one or more METS files can be present in a package. The METS file describing the whole package is from now called "Root METS", or "Package METS" and the METS file in the Representation folders are called "Representation METS". The detailed specification of using METS within the CSIP is available in Section 5.3.

Reviewer 2: Is there a reason why we need two names for the same thing?

Author: Sorry I dwelt on this as I (hope I?) misunderstood the subject as Rep level vs. Package level, but I think I now realise that you mean "Root METS" vs. "Package METS"? If that's the case then I prefer Package METS.

Reviewer: I guess Phillip meant the synonyms:

>from now called "Root METS", or "Package METS"

I agree, we should tame the temptation to make the text more flowery, and instead, stick to one term per concept, wherever possible. I tend to favour "Root METS," but before strongly pushing it, I need to fully familiarise with the changes to structural rules, i.e. I'm afraid that if we allow IPs with no folder structure, "Root" would become meaningless.

Reviewer 2: Yes sorry - I meant: 

> "Root METS", or "Package METS"

I also prefer package METS.

> In addition to the METS files the CSIP recommends the inclusion of PREMIS metadata (PREservation Metadata Implementation Strategies, <http://www.loc.gov/standards/premis/>), included in appropriate preservation metadata folders. This facilitates interoperable approach towards provenance and access to Information Packages. However, we recognise that, especially in the

Reviewer: ```suggestion
In addition to the METS files, the CSIP recommends the inclusion of PREMIS metadata (PREservation Metadata Implementation Strategies, <http://www.loc.gov/standards/premis/>), placed in appropriate preservation metadata folders. This facilitates interoperable approach towards provenance and access to Information Packages. However, we recognise that, especially in the
```

case of SIPs, appropriate preservation metadata is not always available. As such this is also not an absolute requirement though highly desirable. The detailed specification of the use of PREMIS within the CSIP is available in Section 5.4.

The specification does not prohibit the use of additional metadata within Information Packages.

## 5.2 General requirements for the use of metadata

> Before detailing the requirements covering the use of METS and PREMIS we would like to highlight some general metadata concerns which should be handled consistently.

Reviewer: ```suggestion
Before detailing the requirements covering the use of METS and PREMIS, we would like to highlight some general metadata concerns, which should be handled consistently.
```

### The use of identifiers

> Both METS, and by derivation the CSIP, make extensive use of the XML ID data type, see: <https://www.w3.org/TR/xml-id/>. Valid XML IDs must also conform to the NCName restrictions, see: <https://www.w3.org/TR/REC-xml-names/#NT-NCName>, and must begin a letter or an underscore character (‘_’), and contain no characters other than letters, digits, hyphens, underscores, full stops, plus some combination and extension characters. If your package makes use of IDs that don't conform to this specification one mitigating strategy is to add a prefix to all identifiers. For example UUIDs are a common choice for arbitrary identifiers and comprise of hex characters and hyphens, e.g. the value `906F4F12-BA52-4779-AE2C-178F9206111F`. NCName and xml:id values can't legally begin with numeric characters making many UUIDs,

Reviewer: ```suggestion
Both METS, and by derivation the CSIP, make extensive use of the XML ID data type (see: <https://www.w3.org/TR/xml-id/>). Valid XML IDs must also conform to the NCName restrictions (see: <https://www.w3.org/TR/REC-xml-names/#NT-NCName>), and must begin with a letter or an underscore character (‘_’), and contain no characters other than letters, digits, hyphens, underscores, full stops, plus some extension and combination characters.

If a package makes use of IDs that do not conform to this specification, one mitigating strategy is to prefix all identifiers. For example UUIDs are a common choice for arbitrary identifiers and comprise of hex characters and hyphens, e.g. the value `906F4F12-BA52-4779-AE2C-178F9206111F`. NCName and xml:id values cannot legally begin with numeric characters, making many UUIDs, including the example, invalid. Here are two possible solutions, using prefixes:
```

> including the example, invalid. The examples below show two possible solutions, using prefixes:

Reviewer: ```suggestion
```

Examples:

Example 1: The prefix consists of the identifier type acronym and a hyphen: `uuid-`.

```xml
<dmdSec ID="uuid-906F4F12-BA52-4779-AE2C-178F9206111F" CREATED="2018-04-
24T14:37:49.609+01:00">
```

> Example 2: A fixed prefix:`ID`.

Reviewer: ```suggestion
Example 2: A generic prefix: `ID`.
```

`uuid-` in example 1 is also fixed, so we need another word, "generic" is the first that came to mind.

```xml
<dmdSec ID="ID906F4F12-BA52-4779-AE2C-178F9206111F" CREATED="2018-04-
24T14:37:49.609+01:00">
```

> Note that the identifiers specified within the CSIP are mainly used as internal references between Information Package components. No specific prefix is required, other than any IDs are valid `xml:id` types. If a prefix is used we recommend a single prefix is used consistently across all IDs in the package.

Reviewer: ```suggestion
Note that the identifiers specified within the CSIP are mainly used as internal references between Information Package components. Prefixes are not mandatory, but if they are used, we recommend choosing a single prefix and using it consistently across all IDs in the package.
```

`xml:id` restrictions were already explained above.

### Referencing between files within a CSIP Information Package
The CSIP strongly recommends that all components of an information package (i.e. all data, metadata and other parts) are stored as discrete files within the package. While this approach simplifies the management of the Information Package, making it easier to aggregate, validate and modify the package, it also necessitates a clear method for recording and resolving referencing between files.

For example, when using the CSIP METS Profile referencing can occur between:

- descriptive and administrative metadata files described in the `amdSec` and `dmdSec` elements;

> - content (data) files, components of documentation and schemas described in the `fileSec` element;

Reviewer: ```suggestion
- content (data) files, components of documentation, and schemas described in the `fileSec` element;
```

Author: Oxford comma ;)

- representation METS files described in the root METS file’s `fileSec` and `structMap` elements.

A consistent approach to referencing between package components is a fundamental requirement for Information Package validation and integrity checking. Different technical solutions exist for referencing and resolving but they're not universally supported by all digital preservation tools. To ensure interoperability, all references within a CSIP Information Package must adhere the requirements stated in this specification.

### Referencing other packages

> It is important that external references e.g. to other packages are expressed consistently, in the same manner as internal references. All external references MUST USE `mets/@OBJID` attribute of the package.

Reviewer: ```suggestion
It is important that external references (i.e. to other packages) are expressed consistently, in the same manner as internal references. All references to other packages MUST USE the `mets/@OBJID` value of the target package.
```
